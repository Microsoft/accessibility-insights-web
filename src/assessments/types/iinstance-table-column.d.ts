// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { IAssessmentInstanceRowData } from '../../DetailsView/components/assessment-instance-table';

export interface InstanceTableColumn {
    key: string;
    name: string;
    onRender: (item: IAssessmentInstanceRowData) => JSX.Element;
}
