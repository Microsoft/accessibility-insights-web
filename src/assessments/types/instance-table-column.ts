// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { GeneratedAssessmentInstance } from 'common/types/store-data/assessment-result-data';

export interface AssessmentInstanceRowData<P = {}> {
    key: string;
    statusChoiceGroup: JSX.Element;
    visualizationButton?: JSX.Element;
    instance: GeneratedAssessmentInstance<P>;
}

export interface InstanceTableColumn {
    key: string;
    name: string;
    onRender: (item: AssessmentInstanceRowData) => JSX.Element;
}
