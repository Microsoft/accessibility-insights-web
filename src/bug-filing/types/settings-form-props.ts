// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { SettingsDeps } from './../../DetailsView/components/settings-panel/settings/settings-props';

export type SettingsFormProps<Settings> = {
    deps: SettingsDeps;
    settings: Settings;
};
