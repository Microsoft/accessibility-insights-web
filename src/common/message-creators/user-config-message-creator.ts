// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    SetBugServicePayload,
    SetBugServicePropertyPayload,
    SetHighContrastModePayload,
    SetIssueTrackerPathPayload,
    SetTelemetryStatePayload,
} from '../../background/actions/action-payloads';
import { Messages } from '../messages';
import { BaseActionMessageCreator } from './base-action-message-creator';

export class UserConfigMessageCreator extends BaseActionMessageCreator {
    public setTelemetryState(enableTelemetry: boolean): void {
        const payload: SetTelemetryStatePayload = {
            enableTelemetry,
        };

        this.dispatchMessage({
            type: Messages.UserConfig.SetTelemetryConfig,
            tabId: this._tabId,
            payload,
        });
    }

    public setHighContrastMode(enableHighContrast: boolean): void {
        const payload: SetHighContrastModePayload = {
            enableHighContrast,
        };

        this.dispatchMessage({
            type: Messages.UserConfig.SetHighContrastConfig,
            tabId: this._tabId,
            payload,
        });
    }

    public setBugService(bugServiceName: string): void {
        const payload: SetBugServicePayload = {
            bugServiceName,
        };

        this.dispatchMessage({
            type: Messages.UserConfig.SetBugService,
            tabId: this._tabId,
            payload,
        });
    }

    public setBugServiceProperty(bugServiceName: string, propertyName: string, propertyValue: string): void {
        const payload: SetBugServicePropertyPayload = {
            bugServiceName,
            propertyName,
            propertyValue,
        };

        this.dispatchMessage({
            type: Messages.UserConfig.SetBugServiceProperty,
            tabId: this._tabId,
            payload,
        });
    }

    public setIssueTrackerPath(issueTrackerPath: string): void {
        const payload: SetIssueTrackerPathPayload = {
            issueTrackerPath,
        };

        this.dispatchMessage({
            type: Messages.UserConfig.SetIssueTrackerPath,
            tabId: this._tabId,
            payload,
        });
    }
}
