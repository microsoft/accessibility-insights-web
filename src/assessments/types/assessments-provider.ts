// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { VisualizationType } from 'common/types/visualization-type';
import { DictionaryStringTo } from 'types/common-types';
import { Assessment } from './iassessment';
import { Requirement } from './requirement';

export interface AssessmentsProvider {
    all(): ReadonlyArray<Readonly<Assessment>>;
    isValidType(visualizationType: VisualizationType): boolean;
    forType(visualizationType: VisualizationType): Readonly<Assessment>;
    isValidKey(key: string): boolean;
    forKey(key: string): Readonly<Assessment>;
    forRequirementKey(key: string): Readonly<Assessment>;
    getStep(visualizationType: VisualizationType, key: string): Readonly<Requirement>;
    getStepMap(
        visualizationType: VisualizationType,
    ): Readonly<DictionaryStringTo<Readonly<Requirement>>>;
}
