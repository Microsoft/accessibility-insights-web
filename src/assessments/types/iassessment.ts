// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { RequirementOrdering } from '../../common/assessment/requirement';
import { IAssesssmentVisualizationConfiguration } from '../../common/configs/visualization-configuration-factory';
import { AnyExtension } from '../../common/extensibility/extension-point';
import { IAssessmentData } from '../../common/types/store-data/iassessment-result-data';
import { VisualizationType } from '../../common/types/visualization-type';
import { ContentPageComponent } from '../../views/content/content-page';
import { TestStep } from './test-step';

interface BaseAssessment {
    key: string;
    type: VisualizationType;
    title: string;
    gettingStarted: JSX.Element;
    guidance?: ContentPageComponent;
    steps: TestStep[];
    featureFlag?: { required?: string[] };
    executeAssessmentScanPolicy?: (scheduleScan: (step: string) => void, data: IAssessmentData) => void;
    requirementOrder?: RequirementOrdering;
    extensions?: AnyExtension[];
}

export interface ManualAssessment extends BaseAssessment {}

export interface AssistedAssessment extends BaseAssessment {
    storeDataKey: string;
    visualizationConfiguration?: Partial<IAssesssmentVisualizationConfiguration>;
}

export interface Assessment extends BaseAssessment {
    getVisualizationConfiguration: () => IAssesssmentVisualizationConfiguration;
    requirementOrder: RequirementOrdering;
}
