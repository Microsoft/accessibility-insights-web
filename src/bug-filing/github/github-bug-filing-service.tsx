// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { isEmpty, isString } from 'lodash';
import { TextField } from 'office-ui-fabric-react/lib/TextField';
import * as React from 'react';

import { NamedSFC } from '../../common/react/named-sfc';
import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
import { BugFilingService } from '../types/bug-filing-service';
import { SettingsFormProps } from '../types/settings-form-props';

const GitHubBugFilingServiceKey = 'gitHub';

export type GitHubBugFilingSettings = {
    repository: string;
};

function buildStoreData(repository: string): GitHubBugFilingSettings {
    return {
        repository,
    };
}

function isSettingsValid(data: GitHubBugFilingSettings): boolean {
    return !isEmpty(data) && !isEmpty(data.repository) && isString(data.repository) && !isEmpty(data.repository.trim());
}

const renderSettingsForm = NamedSFC<SettingsFormProps<GitHubBugFilingSettings>>('BugFilingSettings', props => {
    const onGitHubRepositoryChange = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
        const propertyName: keyof GitHubBugFilingSettings = 'repository';
        props.deps.userConfigMessageCreator.setBugServiceProperty(GitHubBugFilingServiceKey, propertyName, newValue);
    };

    return (
        <TextField
            className="issue-setting"
            label="Enter desired GitHub repo link:"
            onChange={onGitHubRepositoryChange}
            value={props.settings.repository}
            placeholder="https://github.com/owner/repo"
        />
    );
});

const createBugFilingUrl()

export const GitHubBugFilingService: BugFilingService<GitHubBugFilingSettings> = {
    key: GitHubBugFilingServiceKey,
    displayName: 'GitHub',
    renderSettingsForm,
    buildStoreData,
    isSettingsValid,
    createBugFilingUrl: (data: GitHubBugFilingSettings, bugData: CreateIssueDetailsTextData) => null,
};
