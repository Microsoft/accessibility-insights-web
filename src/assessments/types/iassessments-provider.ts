// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { VisualizationType } from '../../common/types/visualization-type';
import { Assessment } from './iassessment';
import { TestStep } from './test-step';

// tslint:disable-next-line:interface-name
export interface IAssessmentsProvider {
    all(): ReadonlyArray<Readonly<Assessment>>;
    isValidType(type: VisualizationType): boolean;
    forType(type: VisualizationType): Readonly<Assessment>;
    isValidKey(key: string): boolean;
    forKey(key: string): Readonly<Assessment>;
    getStep(type: VisualizationType, key: string): Readonly<TestStep>;
    getStepMap(type: VisualizationType): Readonly<IDictionaryStringTo<Readonly<TestStep>>>;
}
