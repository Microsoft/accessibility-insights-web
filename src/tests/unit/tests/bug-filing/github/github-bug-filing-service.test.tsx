// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { GithubBugFilingService, GithubBugFilingSettings } from '../../../../../bug-filing/github/github-bug-filing-service';
import { Mock, IMock, Times } from 'typemoq';
import { UserConfigMessageCreator } from '../../../../../common/message-creators/user-config-message-creator';
import { SettingsFormProps } from '../../../../../bug-filing/types/settings-form-props';
import { shallow } from 'enzyme';
import * as React from 'react';
import { TextField } from 'office-ui-fabric-react/lib/TextField';

describe('GithubBugFilingServiceTest', () => {
    let userConfigMessageCreatorMock: IMock<UserConfigMessageCreator>;
    let props: SettingsFormProps<GithubBugFilingSettings>;
    beforeEach(() => {
        userConfigMessageCreatorMock = Mock.ofType(UserConfigMessageCreator);
        props = {
            deps: {
                userConfigMessageCreator: userConfigMessageCreatorMock.object,
            },
            settings: {
                repository: 'repo',
            },
        };
    });

    test('static properties', () => {
        expect(GithubBugFilingService.key).toBe('github');
        expect(GithubBugFilingService.displayName).toBe('Github');
    });

    test('buildStoreData', () => {
        const url = 'base';
        const expectedStoreData: GithubBugFilingSettings = {
            repository: url,
        };
        expect(GithubBugFilingService.buildStoreData(url)).toEqual(expectedStoreData);
    });

    test('isSettingsValid', () => {
        const emptySettings: GithubBugFilingSettings = null;
        const invalidSettings: GithubBugFilingSettings = {
            repository: '  ',
        };
        const validSettings: GithubBugFilingSettings = {
            repository: 'repository',
        };
        expect(GithubBugFilingService.isSettingsValid(emptySettings)).toBe(false);
        expect(GithubBugFilingService.isSettingsValid(invalidSettings)).toBe(false);
        expect(GithubBugFilingService.isSettingsValid(validSettings)).toBe(true);
    });

    test('renderSettingsForm', () => {
        const Component = GithubBugFilingService.renderSettingsForm;
        const wrapper = shallow(<Component {...props} />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('renderSettingsForm: onChange', () => {
        const Component = GithubBugFilingService.renderSettingsForm;
        const wrapper = shallow(<Component {...props} />);
        userConfigMessageCreatorMock
            .setup(ucmm => ucmm.setBugServiceProperty('gitHub', 'repository', 'new value'))
            .verifiable(Times.once());
        wrapper
            .find(TextField)
            .props()
            .onChange(null, 'new value');
        userConfigMessageCreatorMock.verifyAll();
    });

    test('fileBug', () => {
        expect(GithubBugFilingService.fileBug).not.toBeNull();
    });
});
