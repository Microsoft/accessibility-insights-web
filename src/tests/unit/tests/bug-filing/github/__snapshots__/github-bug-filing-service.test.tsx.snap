// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubBugFilingServiceTest renderSettingsForm 1`] = `
<React.Fragment>
  <StyledTextFieldBase
    className="issue-setting"
    label="Enter the desired GitHub repo link to file issue:"
    onChange={[Function]}
    placeholder="https://github.com/owner/repo"
    value="repo"
  />
</React.Fragment>
`;
